src/category_theory/abelian/projective.lean

```diff
@@ -74,6 +85,36 @@ end
 lemma iso_iff {P Q : C} (i : P ≅ Q) : projective P ↔ projective Q :=
 ⟨of_iso i, of_iso i.symm⟩
 
+/-- The axiom of choice says that every type is a projective object in `Type`. -/
+instance (X : Type u) : projective X :=
+{ factors := λ E X' f e epi,
+  ⟨λ x, ((epi_iff_surjective _).mp epi (f x)).some,
+  by { ext x, exact ((epi_iff_surjective _).mp epi (f x)).some_spec, }⟩ }
+
+instance Type_enough_projectives : enough_projectives (Type u) :=
+{ presentation := λ X, ⟨{ P := X, f := 𝟙 X, }⟩, }
+
+instance {P Q : C} [has_binary_coproduct P Q] [projective P] [projective Q] :
+  projective (P ⨿ Q) :=
+{ factors := λ E X' f e epi, by exactI
+  ⟨coprod.desc (factor_thru (coprod.inl ≫ f) e) (factor_thru (coprod.inr ≫ f) e), by tidy⟩, }
+
+instance {β : Type v} (g : β → C) [has_coproduct g] [∀ b, projective (g b)] :
+  projective (∐ g) :=
+{ factors := λ E X' f e epi, by exactI
+  ⟨sigma.desc (λ b, factor_thru (sigma.ι g b ≫ f) e), by tidy⟩, }
+
+instance {P Q : C} [has_zero_morphisms C] [has_binary_biproduct P Q]
+  [projective P] [projective Q] :
+  projective (P ⊞ Q) :=
+{ factors := λ E X' f e epi, by exactI
+  ⟨biprod.desc (factor_thru (biprod.inl ≫ f) e) (factor_thru (biprod.inr ≫ f) e), by tidy⟩, }
```

Comment:
It would be nice if this didn't need to be separately stated, let alone proved, and Lean just knew that biproducts were coproducts, but it doesn't seem easy in our current setup. :-(